library/java/net/openid/appauth/AuthorizationService.java

```diff
@@ -213,38 +213,75 @@ public void performAuthorizationRequest(
             @NonNull PendingIntent completedIntent,
             @Nullable PendingIntent canceledIntent,
             @NonNull CustomTabsIntent customTabsIntent) {
-        checkNotDisposed();
-
-        if (mBrowser == null) {
-            throw new ActivityNotFoundException();
-        }
-
-        Uri requestUri = request.toUri();
-        Intent intent;
-        if (mBrowser.useCustomTab) {
-            intent = customTabsIntent.intent;
-        } else {
-            intent = new Intent(Intent.ACTION_VIEW);
-        }
-        intent.setPackage(mBrowser.packageName);
-        intent.setData(requestUri);
-
-        Logger.debug("Using %s as browser for auth, custom tab = %s",
-                intent.getPackage(),
-                mBrowser.useCustomTab.toString());
-        intent.putExtra(CustomTabsIntent.EXTRA_TITLE_VISIBILITY_STATE, CustomTabsIntent.NO_TITLE);
-
-        Logger.debug("Initiating authorization request to %s",
-                request.configuration.authorizationEndpoint);
+        Intent authIntent = prepareAuthorizationRequestIntent(request, customTabsIntent);
         mContext.startActivity(AuthorizationManagementActivity.createStartIntent(
                 mContext,
                 request,
-                intent,
+                authIntent,
                 completedIntent,
                 canceledIntent));
     }
 
     /**
+     * Constructs an intent that encapsulates the provided request and custom tabs intent,
+     * and is intended to be launched via {@link Activity#startActivityForResult}
+     * When started, the intent launches an {@link Activity} that sends an authorization request
+     * to the authorization service, using a
+     * [custom tab](https://developer.chrome.com/multidevice/android/customtabs).
```

Comment:
There's probably a little too much detail here actually, and it can be misleading as a custom won't always be used (this depends on availability). I would just omit this sentence.